File: frontend/lib/models/account.dart
class Account {
  final String id;
  final String name;
  final String currencyCode;
  final double currentBalance;
  final double? availableBalance;
  final DateTime balanceDate;
  final String organizationDomain;
  final String? accountType;
  final DateTime createdAt;

  Account({
    required this.id,
    required this.name,
    required this.currencyCode,
    required this.currentBalance,
    this.availableBalance,
    required this.balanceDate,
    required this.organizationDomain,
    this.accountType,
    required this.createdAt,
  });

  factory Account.fromJson(Map<String, dynamic> json) {
    return Account(
      id: json['id'],
      name: json['name'],
      currencyCode: json['currency_code'],
      currentBalance: json['current_balance'].toDouble(),
      availableBalance: json['available_balance']?.toDouble(),
      balanceDate: DateTime.parse(json['balance_date']),
      organizationDomain: json['organization_domain'],
      accountType: json['account_type'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}

File: server/app/schemas/account.py

from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from enum import IntEnum


# Enum for account type (schema-only)
class AccountType(IntEnum):
    CHECKING = 0
    SAVINGS = 1
    CREDIT = 2
    INVESTMENT = 3
    LOAN = 4
    OTHER = 5

class AccountBase(BaseModel):
    name: str

class AccountResponse(AccountBase):
    id: str
    currency_code: str
    current_balance: float
    available_balance: Optional[float] = None
    balance_date: datetime
    organization_domain: str
    account_type: AccountType
    created_at: datetime

    class Config:
        from_attributes = True

        
class ChangeTypeRequest(BaseModel):
    new_type: int
